dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -10,26 +10,81 @@ sealed trait UntypedExpression[T] {
   def expr: Expression
 }
 
```

Comment:
@OlivierBlanvillain from the gitter discussion, you at the type of res3. The type of the dataframe should be Long but it shows `frameless.TypeCoercion.intLong.ResultType` instead. If you go into the implementation of `intLong` you will see that that ResultType = Long.